test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -250,11 +248,23 @@ func (sm *StackManager) getStack(ctx context.Context, name string, config runner
 		logger = logWriter
 	}
 
-	upResult, err := stack.Up(upCtx, optup.ProgressStreams(logger), optup.DebugLogging(debug.LoggingOptions{
-		FlowToPlugins: true,
-		LogLevel:      &loglevel,
-	}))
-
+	var ok bool
+	var upResult auto.UpResult
+
+	for retry := 0; retry < 2 && !ok; retry++ {
```

Comment:
`2` should be a const.

Also we should produce telemetry to identify the test cases producing these errors to fix them.

Reply:
Changed the number of retry to a const.

What do you mean by telemetry? I currently log a message in the test case saying that a Pulumi error that should be retried was raised and that the pulumi up will be retried. Do you think we need something else?

Reply:
It would be nice to have the info in the CI App, but I don't know if it's possible. Otherwise you could submit a Datadog Event.

Reply:
Sounds good, I will work on this so we can add some info about retries on the E2E tests dashboard